### PR TITLE
[CN-585] Add operator chart and update make file

### PR DESCRIPTION
**Pull request by cagric0**
## Description

Add operator helm chart 
Add new make commands
Update `tilt` file for using updated make commands
Replace `NAME_PREFIX` with `RELEASE_NAME` in the test workflows
Remove `bundle.yaml`

## User Impact

`bundle.yaml` is removed and helm is promoted for deploying the operator

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -44,6 +44,5 @@ All PRs follow the same process:
 3.	The latest changes are in the `main` branch.
 4.	Make sure to design clean commits that are easily readable. That includes descriptive commit messages.
 5.  Please keep your PRs as small as possible, _i.e._ if you plan to perform a huge change, do not submit a single and large PR for it. For an enhancement or larger feature, you can create a GitHub issue first to discuss.
-6.  Run `make generate-bundle-yaml` command, if you modify the API of Custom Resources.
-7.  Before you push, run the command `make lint` in your terminal and fix the linter errors if any. Push your PR once it is free of linter errors.
-8.	If you submit a PR as the solution to a specific issue, please mention the issue number either in the PR description or commit message.
\ No newline at end of file
+6.  Before you push, run the command `make lint` in your terminal and fix the linter errors if any. Push your PR once it is free of linter errors.
```
Comment (cheels) [inline CONTRIBUTING.md:47]: Nothing about what to do if contributor wants to modify the API of Custom Resourses and how it should be reflected with our new structure (with Helm). Is it ok?

**Review by dzeromski-hazelcast — APPROVED** [review]

File: helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v2
+
+name: hazelcast-platform-operator-crds
+
+description: A Helm chart for installing the Hazelcast Platform Operator Custom Resource Definitions (CRDs).
+
+type: application
+
+version: 5.0.6-alpha
```
Comment (hasancelik) [inline helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/Chart.yaml:9]:
```suggestion
version: 5.0.6-rc
```

File: helm-charts/hazelcast-platform-operator/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: v2
+
+name: hazelcast-platform-operator
+
+description: A Helm chart for installing Hazelcast Platform Operator which automates common management tasks such as configuring, creating, scaling, and recovering Hazelcast clusters on Kubernetes and Red Hat OpenShift. By taking care of manual deployment and life-cycle management, Hazelcast Platform Operator makes it simpler to work with Hazelcast clusters.
+
+type: application
+
+version: 5.0.6-alpha
```
[the remark this answers is not in the document]
Comment (hasancelik) [reply helm-charts/hazelcast-platform-operator/Chart.yaml:9]: And also, why are we using `5.0.6`, not `5.6`?

**Review by hasancelik — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -81,8 +83,17 @@ endif
...
+OPERATOR_CHART ?= ./helm-charts/hazelcast-platform-operator
+CRD_CHART := $(OPERATOR_CHART)/charts/hazelcast-platform-operator-crds
+
 PHONE_HOME_ENABLED ?= false
 DEVELOPER_MODE_ENABLED ?= true
+INSTALL_CRDS ?= false
+
+RELEASE_NAME ?= v1
+CRD_RELEASE_NAME ?= hazelcast-platform-operator-crds
+DEPLOYMENT_NAME := $(RELEASE_NAME)-hazelcast-platform-operator
```
Comment (mtyazici) [inline Makefile:95]: I did not try this case but, DEPLOYMENT_NAME might not work in operators deployed with OLM.

Comment (cagric0) [reply Makefile:96]: We are using this field only in the test env. In `deployment.yaml` we are still passing POD_NAME

**Pull request merged**